elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/AggregationDataStoreDynamicSecurityIT.java

```diff
@@ -0,0 +1,162 @@
+        DataSource defaultDataSource = new HikariDataSource(config);
+        String defaultDialect = "h2";
+        ConnectionDetails defaultConnectionDetails = new ConnectionDetails(defaultDataSource, defaultDialect);
+
+        Properties prop = new Properties();
+        prop.put("javax.persistence.jdbc.driver", config.getDriverClassName());
+        prop.put("javax.persistence.jdbc.url", config.getJdbcUrl());
+        EntityManagerFactory emf = Persistence.createEntityManagerFactory("aggregationStore", prop);
+
+        Map<String, ConnectionDetails> connectionDetailsMap = new HashMap<>();
+        // Add connection details fetched from hjson
+        connectionDetailsMap.putAll(compiler.getConnectionDetailsMap());
+
+        return new AggregationDataStoreTestHarness(emf, defaultConnectionDetails, connectionDetailsMap, compiler);
+    }
+
+    @Test
+    public void testSecurityHjosnWithAggregationModel() throws Exception {
+
+        SecurityContextUser operator = new SecurityContextUser(new SecurityContext() {
+            @Override
+            public boolean isUserInRole(String role) {
+                switch (role) {
+                    case "admin":
+                        return false;
+                    case "operator":
+                        return true;
+                    case "guest user":
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+
+            @Override
+            public boolean isSecure() {
+                return false;
+            }
+
+            @Override
+            public Principal getUserPrincipal() {
+                return () -> "1";
+            }
+
+            @Override
+            public String getAuthenticationScheme() {
+                return null;
+            }
+        });
+
+        SecurityContextUser admin = new SecurityContextUser(new SecurityContext() {
+            @Override
+            public boolean isUserInRole(String role) {
+                switch (role) {
+                    case "admin":
+                        return true;
+                    case "operator":
+                        return true;
+                    case "guest user":
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+
+            @Override
+            public boolean isSecure() {
+                return false;
+            }
+
+            @Override
+            public Principal getUserPrincipal() {
+                return () -> "1";
+            }
+
+            @Override
+            public String getAuthenticationScheme() {
+                return null;
+            }
+        });
+
+        EntityDictionary dictionary = new EntityDictionary(new HashMap<>());
+        dictionary.addSecurityChecks(compiler.findAnnotatedClasses(SecurityCheck.class));
+
+        Elide elide = new Elide(new ElideSettingsBuilder(dataStore)
```

Comment:
Rather than testing at a lower level, you could look at TestAuthFilter and how it is used.  It might make sense to extend that in some way with a mock that can be controlled in the tests themselves.  Then you could also use Rest Assured here for more readable tests.